vt/feed.py

```diff
@@ -130,46 +127,35 @@ async def _get_next_batch_async(self):
         if missing_batches > self._missing_batches_tolerancy:
           raise error
 
-  def _get_next_batch(self):
-    return asyncio.get_event_loop().run_until_complete(
-        self._get_next_batch_async())
-
   def _skip(self, n):
     for _ in range(n):
       self._batch.readline()
       self._batch_cursor += 1
 
   def __iter__(self):
-    while True:
-      if self._batch:
-        next_item = self._batch.readline()
-      else:
-        self._get_next_batch()
-        self._skip(self._batch_skip)
-        self._batch_skip = 0
-        next_item = self._batch.readline()
-      self._batch_cursor += 1
-      self._count += 1
+    return self
 
-      if next_item:
-        yield Object.from_dict(json.loads(next_item.decode('utf-8')))
-      else:
-        self._batch = None
+  def __aiter__(self):
+    return self
 
-  async def __aiter__(self):
+  def __next__(self):
+    try:
+      return make_sync(self.__anext__())
+    except StopAsyncIteration:
```

Comment:
The try/except is useless for the feed since iteration doesn't stop. Kept it for symmetry with the Iterator's implementation